src/pages/SparkPage.tsx

```diff
@@ -143,9 +144,13 @@ function Dashboard() {
         setWindowType(2)
     }
 
+    function handleLogin(name: string){
+        setUserName(name);
+    }
+
     return (
         <div className="sparkBackground" id="sparkPageTop">
-            {displayPost && <Popup windowType={windowType} hidePost={hidePost} />}
+            {displayPost && <Popup windowType={windowType} hidePost={hidePost} setName = {handleLogin}/>}
```

Comment:
hmm why is setName equalling handleLogin? It's a good idea for names to match each other (and describe exactly what they do). This is a little confusing